### ISSUE TITLE
Error correction is not supported (or failed) with Samsung SE-506CB drive

### ISSUE THREAD

**Issue by sanyifashion** [speed47/qpxtool#17]
The quality scanning feature is supported by the external blu ray burner Samsung SE-506CB, yet it does not work with QPxTool, tested with a BD-XL (TL) disc:

![image](https://user-images.githubusercontent.com/47088572/95782213-ac18c480-0ccf-11eb-95e9-eec51c967b92.png)

**Comment by speed47** [comment]
The TSSTcorp drives are explicitly blocked using the generic "LiteOn" plugin, by the original author, I don't know why.
The HL-DT-ST drives were also blocked, and it was preventing me using my WH16NS48 drive, which does work correctly with this plugin. So I removed the block.

The corresponding line of code is this one:
https://github.com/speed47/qpxtool/blob/master/plugins/liteon/qscan_plugin.h#L127

Can you compile programs? If yes, you might want to give it a try, removing this line. I'll be interested in the results, of course.
If not, and if you want to try it, I may be able to provide you a special ^recompiled version so you can test.

Of course, obligatory reminder here: I don't know why these drives were blocked in the plugin, my guess is that the author thinks all these drives are just not compatible so to avoid false positives, it just denies them directly in the code. However nowadays will all the rebranded drives everywhere, this is proving wrong. At least this was the case with my drive. But I may be completely wrong and maybe the LiteOn sequence of commands was at some point known to cause problems with some drives. I can't know for sure, and my guess is that commands shouldn't hurt drives, but it's not my drive, it's yours!

**Comment by speed47** [comment]
No worries, I can put up a precompiled version somewhere for you easily, as I have all the toolchain in place, it's not too much of a hassle.
I agree with you, I think there is no risk in testing those read commands, if it was my drive, I would try it (that's what I did, successfully, with my HL-DT-ST drive).

Here is the `libqscan_liteon.dll` file with the TSSTcorp block removed:
[libqscan_liteon.zip](https://github.com/speed47/qpxtool/files/5370730/libqscan_liteon.zip)

You can put it in the `pl3` folder, and see if it works better with your drive.

If it does, I'll publish a `pl4` with this change.

**Comment by sanyifashion** [comment]
Yeah I got an LG drive too, which I crossflashed to WH16NS58. I was afraid I would wreck the drive but it worked and scans perfectly. And the same goes for your fix with the Samsung:

![image](https://user-images.githubusercontent.com/47088572/95882256-c313f100-0d79-11eb-92cf-e593f118555a.png)

Thank you, that was all I wanted ;) My only concern is that I have set the speed to 4x and as you can see on the graph it stayed on 2,4x or something. Maybe it's a limitation of the drive, I'll try 5x next time, because that was the maximum I could set.

**Comment by speed47** [comment]
> Yeah I got an LG drive too, which I crossflashed to WH16NS58. I was afraid I would wreck the drive but it worked and scans perfectly.

Exactly what I did too, crossflashed to WH16NS58 firmware 1.V5.
If you have some spare time and a CD somewhere, could you try to scan it? On my drive the behaviour is super strange, the scan goes back and forth (it's easier to see on the console qscan program that qpxtool uses on the background), and aborts before the end of the scan. DVD and BD scans are okay however.

> Thank you, that was all I wanted ;) My only concern is that I have set the speed to 4x and as you can see on the graph it stayed on 2,4x or something. Maybe it's a limitation of the drive, I'll try 5x next time, because that was the maximum I could set.

Perfect, thanks for your feedback! I'm releasing `pl4` with the change.

**Comment by sanyifashion** [comment]
I found a CD and tried to test it with my LG but it was very strange, like it got into some infinite loop and the scan would go forever... 

The graph was also not drawn properly beyond the ~7th minute:
![image](https://user-images.githubusercontent.com/47088572/95954060-08c4ce00-0dfb-11eb-9d1c-7da437ae8d96.png)

I scanned the same disc with my Lite-On and it was done in ~4 minutes without any problem.

**Comment by speed47** [comment]
You're having the same issue as I do on this drive. I think this might be a firmware problem (only on CD scanning), because we have exactly the same drive and firmware.
What happens is more clear on the console version of the tool, but the drive constantly jumps back to previous sections of the disc it already has scanned, that's why the graph seems completely borked.
I'm not sure I've tried scanning a CD on this drive with other tools however, just to be sure it's not a bug in the program (which seems very unlikely). Will do that.

**Comment by sanyifashion** [comment]
Okay, I did try it with OptiDriveControl but it couldn't even spin up the drive and was completely frozen, it even froze the the explorer I had to restart and reset the computer... I'll try DiscSpeed later if I can unblock the LG drive.

**Comment by speed47** [comment]
I tried with OptiDriveControl and an old version of CD Speed, here they react as if the drive didn't supported CD scanning at all.

**Comment by intracube-paul** [comment]
Hi, can anyone confirm if these builds are likely to compatible with an LG drive on WH16NS60 v1.02mk firmware - including the reporting of correctable and uncorrectable disc errors?

Do these drives also report jitter statistics?

Thanks.

**Comment by speed47** [comment]
Not sure this firmware can report errors, only thing I can say for sure is that my BH16NS55 crossflashed to a BH16NS58-DUP (firmware 1V.5) does report errors with both DVD and BD-R. For CD-Rs, this firmware doesn't report errors correctly, however (as discussed above).
Drives reporting errors are starting to be hard to find nowadays.

**Comment by intracube-paul** [comment]
> Not sure this firmware can report errors, only thing I can say for sure is that my BH16NS55 crossflashed to a BH16NS58-DUP (firmware 1V.5) does report errors with both DVD and BD-R.

Ah ok. If I go ahead and get the LG BH16NS55 with WH16NS60 firmware and manage to get it working, I'll report back. It's a shame there isn't an up to date drive compatibility list.

I've just managed to get the official version of QPxTool (0.8.1) working with my Plextor BD-R PX-B950SA drive (purchased in 2012 and with the original 1.02 firmware). A forum post mentioned that QPxTool neeeds to be run as root to access the error correction feature and sure enough, running as root made this option available (with qscan plugin option left as autodetect). Just thought I'd mention it here in case anyone else made the same mistake.

I've also run into this issue on the official release https://github.com/speed47/qpxtool/issues/3 so another reason to give this branch a try.

**Comment by speed47** [comment]
I've added a text tip in the "run test" dialogbox about the "you might need to run this as root to get error correction scanning feature" in the dev version https://github.com/speed47/qpxtool/releases/tag/devel (yes, I know, 4 years later... but better late than never, right ? :) )
